pepper-apis/src/main/java/org/broadinstitute/ddp/service/Auth0LogEventService.java

```diff
@@ -0,0 +1,112 @@
+ *     <li>adds extracted log events to log (together with tenant name detected from POST URL).</li>
+ * </ul>
+ * NOTE: see Auth0LogEventServiceTest for examples of Auth0 log events JSON docs.
+ */
+public class Auth0LogEventService {
+
+    private final Logger log = getLogger(this.getClass());
+
+    /**
+     * List of Auth0 Log Event JSON node names which to read (recursively)
+     */
+    static final List<String> AUTH0_LOG_EVENT_NODE_NAMES = List.of(
+            LOG_ID.nodeName(),
+            DATE.nodeName(),
+            TYPE.nodeName(),
+            CLIENT_ID.nodeName(),
+            CONNECTION_ID.nodeName(),
+            USER_ID.nodeName(),
+            USER_AGENT.nodeName(),
+            IP.nodeName(),
+            EMAIL.nodeName(),
+            USER_NAME.nodeName(),
+            DATA.nodeName()
+    );
+
+    /**
+     * Parses JSON doc (coming as a payload in POST request specified in Auth0 Custom Webhook)
+     * extracting log events data from it. Data fetched according to names
+     * defined in {@link #AUTH0_LOG_EVENT_NODE_NAMES}.
+     *
+     * @param logEventsJson payload doc coming with POST request (from Auth0 stream).
+     * @return {@code List<Map<String, JsonElement>>} list of maps with parsed log events data
+     */
+    public List<Map<String, JsonElement>> parseAuth0LogEvents(String logEventsJson) {
+        var rootNode = new Gson().fromJson(logEventsJson, JsonObject.class);
+        var logsNode = rootNode.getAsJsonArray(LOGS.nodeName());
+        final List<Map<String, JsonElement>> logEvents = new ArrayList<>();
+        for (JsonElement node : logsNode) {
+            logEvents.add(GsonRecursiveReader.readValues(node, AUTH0_LOG_EVENT_NODE_NAMES));
+        }
+        return logEvents;
+    }
+
+    public void logAuth0LogEvent(final Map<String, JsonElement> logEvent, String tenant) {
+        log.info(
+                "AUTH0-LOG-EVENT[{}]: type:{}, date:{}, log_id:{}\n"
+                        + "\tclient_id:{}, connection_id:{}, user_id:{}\n"
+                        + "\tuser_agent:{}\n"
+                        + "\tip:{}, email:{}\n"
+                        + "\tdata: {}",
+                tenant,
+                resolveStringValue(TYPE, logEvent),
+                resolveDateTimeValue(DATE, logEvent),
+                resolveStringValue(LOG_ID, logEvent),
+                resolveStringValue(CLIENT_ID, logEvent),
+                resolveStringValue(CONNECTION_ID, logEvent),
+                resolveStringValue(USER_ID, logEvent),
+                resolveStringValue(USER_AGENT, logEvent),
+                resolveStringValue(IP, logEvent),
+                resolveStringValue(EMAIL, logEvent),
+                resolveStringValue(DATA, logEvent));
+    }
+
+    /**
+     * Save log event and tenant data to DB table 'auth0-log-event'
+     */
+    public void persistAuth0LogEvent(final Map<String, JsonElement> logEvent, String tenant) {
+        TransactionWrapper.useTxn(handle -> {
```

Comment:
Can we pass in a database `Handle` instead? It'll make things a bit easier to test, and allows for easier composition with other operations in a single transaction.

Reply:
Sure. Passed Handle